### DISCUSSION TITLE
lolMiner 1.39

### DISCUSSION THREAD

**Discussion by Lolliedieb** [Lolliedieb/lolMiner-releases#897] [General]
# lolMiner 1.39
(Windows build is in the testing and will come soon)

### Feature Changes
- Added support for mining TON. Use **--algo TON** to mine it. Developer fee is 1.5%, AMD cards since GCN1 (via OpenCL) and Nvidia cards Maxwell and newer generation (via Cuda) are supported. 
- Added support for mining Ubiq. Use **--algo UBQHASH** to mine it. The fee is 0.7% as with all Ethash based coins.
- Added support for routing your stratum traffic through a socks5 proxy server. Use --socks5 <proxy ip>:<proxy port> to enable it. Note: at the moment this does not work with https traffic as in the TON algorithm. We will work on that.

### Fixes 
- Added a timeout for DOH requests, so they can no longer hang indefinitely.
- Fixed ethash support for RX 5500 series GPUs on more recent drivers. 

### Notes about TON pool connection
At the moment only the https pool communication protocol is supported. Therefore your --pool address must start with https://.   This protocol is not ideal for mining, but we work on establishing a real stratum protocol with some pools in the near future, stay tuned! 
Note when using a pool address starting with https:// specifying a port is optional (the default 443 is used when not specifying one).  Also when mining TON via https lolMiner automatically uses TLS 1.2 encryption with skipped host name verification. As a final notice: always make sure your TON pool either does not need miner wallet registration or you registered - many popular pools require to register your wallet address to mine it.

### Notes about TON performance and tuning
With regard to tuning for TON: The algorithm needs core, core and more core. Therefore maximizing the core clock within the limits of you cooling and energy supply is the key for the ideal performance. It might be beneficial to reduce the core voltage to produce less heat and less energy consumption - this way the cards keep more distance to their thermal limits and can often clock higher as well. We recommend this algorithm on your AMD RX 6000 cards!

 

<hr /><em>This discussion was created from the release <a href='https://github.com/Lolliedieb/lolMiner-releases/releases/tag/1.39'>lolMiner 1.39</a>.</em>

**Comment by rashrrr** [comment]
miner_restarted_after_error on RaveOS
Rig - 6 cards Radeon RX6800
TON coin mining

and many rejected shares

`Dec 31 11:43:02 rig user.info kernel: [254440.075473] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.075477] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.075868] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:02 rig user.info kernel: [254440.137610] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.137614] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.137997] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:02 rig user.info kernel: [254440.196335] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.196340] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.196731] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:02 rig user.info kernel: [254440.265367] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.265371] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.265751] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:02 rig user.info kernel: [254440.322670] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.322674] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.323069] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:02 rig user.info kernel: [254440.384317] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 11:43:02 rig user.err kernel: [254440.384321] NVRM: No NVIDIA GPU found.
Dec 31 11:43:02 rig user.info kernel: [254440.384732] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 11:43:16 rig authpriv.info dropbear[20267]: Child connection from 127.0.0.1:47280
Dec 31 11:43:16 rig authpriv.notice dropbear[20267]: Password auth succeeded for 'root' from 127.0.0.1:47280
Dec 31 11:44:18 rig authpriv.info dropbear[19631]: Exit (root) from &lt;127.0.0.1:46492&gt;: Exited normally
Dec 31 11:44:18 rig authpriv.warn dropbear[19631]: syslogin_perform_logout: logout(pts/2) returned an error: No such process
Dec 31 11:45:10 rig authpriv.info dropbear[20267]: Exit (root) from &lt;127.0.0.1:47280&gt;: Exited normally
Dec 31 11:45:10 rig authpriv.warn dropbear[20267]: syslogin_perform_logout: logout(pts/1) returned an error: No such process
Dec 31 12:03:32 rig daemon.info udhcpc[440]: sending renew to 192.168.0.1
Dec 31 12:03:32 rig daemon.info udhcpc[440]: lease of 192.168.0.105 obtained, lease time 7200
Dec 31 12:30:24 rig user.info kernel: [257282.029778] lolMiner[24987]: segfault at 10 ip 00007f8238377df8 sp 00007f81977fd448 error 6 in libc-2.31.so[7f8238243000+142000]
Dec 31 12:30:24 rig user.info kernel: [257282.029794] Code: d1 f3 a4 c3 80 fa 10 73 17 80 fa 08 73 27 80 fa 04 73 33 80 fa 01 77 3b 72 05 0f b6 0e 88 0f c3 c5 fa 6f 06 c5 fa 6f 4c 16 f0 &lt;c5&gt; fa 7f 07 c5 fa 7f 4c 17 f0 c3 48 8b 4c 16 f8 48 8b 36 48 89 4c
Dec 31 12:30:28 rig user.info kernel: [257286.220952] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.220957] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.221375] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.287723] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.287728] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.288120] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.341735] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.341740] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.342143] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.406661] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.406666] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.407058] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.464979] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.464984] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.465383] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.528523] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.528527] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.528911] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:28 rig user.info kernel: [257286.585697] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:28 rig user.err kernel: [257286.585701] NVRM: No NVIDIA GPU found.
Dec 31 12:30:28 rig user.info kernel: [257286.586077] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.651619] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.651624] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.652006] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.711748] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.711753] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.712127] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.774012] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.774017] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.774402] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.830004] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.830009] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.830426] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.895831] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.895835] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.896235] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257286.951814] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257286.951819] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257286.952222] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.014669] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.014673] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.015052] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.072736] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.072741] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.073137] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.134558] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.134562] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.134945] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.192770] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.192774] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.193158] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.254597] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.254601] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.254975] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.315863] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.315867] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.316255] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:30:29 rig user.info kernel: [257287.379520] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:30:29 rig user.err kernel: [257287.379525] NVRM: No NVIDIA GPU found.
Dec 31 12:30:29 rig user.info kernel: [257287.379927] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:10 rig user.info kernel: [257388.592087] lolMiner[25268]: segfault at 10 ip 00007f8421a6edf8 sp 00007f8386ffc0b8 error 6 in libc-2.31.so[7f842193a000+142000]
Dec 31 12:32:10 rig user.info kernel: [257388.592097] Code: d1 f3 a4 c3 80 fa 10 73 17 80 fa 08 73 27 80 fa 04 73 33 80 fa 01 77 3b 72 05 0f b6 0e 88 0f c3 c5 fa 6f 06 c5 fa 6f 4c 16 f0 &lt;c5&gt; fa 7f 07 c5 fa 7f 4c 17 f0 c3 48 8b 4c 16 f8 48 8b 36 48 89 4c
Dec 31 12:32:18 rig user.info kernel: [257396.292193] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.292198] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.292637] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:18 rig user.info kernel: [257396.363288] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.363293] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.363716] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:18 rig user.info kernel: [257396.419546] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.419550] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.419939] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:18 rig user.info kernel: [257396.485470] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.485475] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.485854] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:18 rig user.info kernel: [257396.541709] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.541713] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.542094] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:18 rig user.info kernel: [257396.603476] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:18 rig user.err kernel: [257396.603480] NVRM: No NVIDIA GPU found.
Dec 31 12:32:18 rig user.info kernel: [257396.603859] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.659671] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.659676] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.660067] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.717751] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.717756] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.718166] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.774651] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.774655] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.775060] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.842402] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.842407] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.842791] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.898651] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.898655] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.899057] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257396.961593] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257396.961597] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257396.961974] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.018886] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.018892] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.019282] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.085612] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.085616] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.085998] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.142677] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.142682] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.143059] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.210499] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.210504] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.210890] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.266568] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.266572] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.266974] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.332571] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.332576] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.332963] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.388949] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.388954] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.389343] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244
Dec 31 12:32:19 rig user.info kernel: [257397.450737] nvidia-nvlink: Nvlink Core is being initialized, major device number 244
Dec 31 12:32:19 rig user.err kernel: [257397.450741] NVRM: No NVIDIA GPU found.
Dec 31 12:32:19 rig user.info kernel: [257397.451128] nvidia-nvlink: Unregistered the Nvlink Core, major device number 244`

**Comment by rashrrr** [comment]
Same problem on RX570

**Comment by dawidmosk** [comment]
ubq nice add but maybe ryo in the future... :)

**Comment by oiG8Uchi** [comment]
![image](https://user-images.githubusercontent.com/95996550/147834558-4daa05cb-ae98-4586-aa37-93a17fe92370.png)
![image](https://user-images.githubusercontent.com/95996550/147834567-768c6e84-b534-4571-bcf0-fb6bdbe62dde.png)
Average speed (15s): 0.00 MH/s

**Comment by Lolliedieb** [comment]
I am investigating that. Seems like a display bug, because the minute stats
still work and the miner still submit shares.
Restart miner to get it back working

oiG8Uchi ***@***.***> schrieb am Fr., 31. Dez. 2021, 18:50:

**Reply by primojones107** [reply]
Hi,

I am testing the TON mining feature on about 500 GPUS right now.
Very often rigs crash with the following:
[  757.042764] lolMiner[1577]: segfault at 10 ip 00007ffbf40790f8 sp 00007ffb8de7f538 error 6 in libc-2.27.so[7ffbf3f0b000+1e7000]
Tried from no OC to agressive OC to moderate, different SMOS images, different drivers. Its all AMD cards 470/570 4gb
Crashes per machine can range from 0 to 30 in a day.
Machines are 6-12 cards each.
On one 3 card machine there is no error 2 days now.

**Comment by i8088** [comment]
Too many reboot last days on TON mining.    

/root/xminer.sh: line 189: 10724 Segmentation fault      ${minerSudo} /root/miner/${MINER_PKG_NAME}/${MINER_FILE} ${MINER_OPTIONS_GO}
Miner ended or crashed. Restarting miner in 30 seconds...
Rebooting... (GPU problem detected). Check crash logs.
Possible cause: Too much undervolt, overclock or broken riser.

**Comment by clownel** [comment]
error....and can be added http too? no secured conection? thx !
Connecting to pool...
DNS over HTTPS resolve failed - switching to standard resolve
terminate called after throwing an instance of 'boost::wrapexcept<boost::system::system_error>'
  what():  resolve: Host not found (authoritative)
Aborted (core dumped)

**Comment by Greenguy36** [comment]
![image](https://user-images.githubusercontent.com/96999778/147889953-5dd0b6ea-c8bd-4a3e-af3a-bce2f5273c8e.png)


any idea on how to fix this?

**Comment by glennch** [comment]
![lolminer ton bug](https://user-images.githubusercontent.com/97136741/148143278-ad5b3cb7-a3d7-4b17-87b3-bc1dbce430e9.jpg)

what does this mean?

**Reply by papnikol** [reply]
I have the same problem. Seems to be what @mohsenk94 said

**Reply by mohsenk94** [reply]
I'm trying out Lol 1.4 now... they seem to be trying to get it to work with more pools + adopt stratum in anyway possible... 
Also the 33.3% fee reduction is nice.
Trying out different pools to see how it goes.. will let you know

**Reply by glennch** [reply]
its OK now guys, you just need to change the server. Try one of these:

Whales have 3 servers.
https://server1.whalestonpool.com
https://pool.services.tonwhales.com
https://ton-pool-server-p3agi.ondigitalocean.app

**Comment by mohsenk94** [comment]
@glennch This happens very often for me
In fact it happened so much so I changed my pool to ton-pool (Yes they have been hacked recently... but they are getting the funds back and paying out people and it is the ONLY one available to mine with LOL )
It has stuff to do with them connecting via CloudFlare and CloudFlare blocking your access...
I am also constantly getting disconnected from ton-pool as well.. IT has to be the HTTPS connections and these pools not being real Stratums yet...
Hopefully they release proper stratum soon...

**Comment by mohsenk94** [comment]
sadly I've been getting all sorts of errors all day the last 2 days and it's affecting my income significantly.
As much as 40% less than it should be!
I reckon till there is a stable stratum , it's essentially impossible to be more profitable than eth on this coin...

**Comment by liaozhijie** [comment]
Hi，i have some questions: 
1. Use 1.42 mining TON has many reject share on whalesPOOL
2. Is TON mining  fee connection encrypted? (TLS)
3. Could lolminer add SNI-hidden?
Thanks !

**Comment by vocals100** [comment]
Hi, maybe you can help me, I do not understand too much how to do this: how do you specify an ETH wallet for receiving TONCOIN mined tokens? I was told that it can be used that way because TONCOIN is an ERC20 token. Thank you.
